File: Leetcode/Java/Q268_Missing_Number.java
class Solution {
    public int missingNumber(int[] nums) {
        // Get the length of the array
        int n = nums.length;

        // Initialize result to n, which is the upper bound of the range [0, n]
        int result = n;

        // Iterate through the array, using XOR to cancel out all numbers except the missing one
        for (int i = 0; i < n; i++) {
            // XOR the current result with both the index 'i' and the value at 'nums[i]'
            result = result ^ i ^ nums[i];
        }

        // Return the result, which will be the missing number
        return result;
    }
}

File: Leetcode/Python/Q268_Missing_Number.py
class Solution(object):
    def missingNumber(self, nums):
        # Get the length of the array
        n = len(nums)
        
        # Iterate through the range from 0 to n (inclusive)
        for i in range(0, n+1):
            # If the current number 'i' is present in the array, continue to the next iteration
            if i in nums:
                continue
            else:
                # If 'i' is not found in the array, return it as the missing number
                return i
